src/streamer.rs

```diff
@@ -458,6 +458,45 @@ pub fn default_window() -> Window {
     Arc::new(RwLock::new(vec![None; WINDOW_SIZE as usize]))
 }
 
+/// Initialize a rebroadcast window with most recent Entry blobs
+/// * `blobs` - up to WINDOW_SIZE most recent blobs
+/// * `entry_height` - current entry height
+pub fn initialized_window(
+    crdt: &Arc<RwLock<Crdt>>,
+    blobs: VecDeque<SharedBlob>,
+    entry_height: u64,
+) -> Window {
+    let window = default_window();
+
+    {
+        let mut win = window.write().unwrap();
+        assert!(blobs.len() <= win.len());
+
+        // flatten deque to vec
+        let mut blobs: Vec<_> = blobs.into_iter().collect();
```

Comment:
You can push this on the caller. Maybe the author of the caller will conclude its vector should be changed to `Vec`.